presto-verifier/src/main/java/com/facebook/presto/verifier/Validator.java

```diff
@@ -296,59 +296,55 @@ private boolean checkForDeterministicAndRerunTestQueriesIfNeeded()
 
     private QueryResult executeQueryTest()
     {
-        Query query = queryPair.getTest();
-        QueryResult queryResult = new QueryResult(State.INVALID, null, null, null, null, ImmutableList.of());
-        try {
-            // startup
-            queryResult = setup(query, testPreQueryResults, testPrequery -> executeQuery(testGateway, testUsername, testPassword, queryPair.getTest(), testPrequery, testTimeout, sessionProperties));
-
-            // if startup is successful -> execute query
-            if (queryResult.getState() == State.SUCCESS) {
-                queryResult = executeQuery(testGateway, testUsername, testPassword, queryPair.getTest(), query.getQuery(), testTimeout, sessionProperties);
-            }
-        }
-        finally {
-            int retry = 0;
-            QueryResult tearDownResult;
-            do {
-                tearDownResult = tearDown(query, testPostQueryResults, testPostquery -> executeQuery(testGateway, testUsername, testPassword, queryPair.getTest(), testPostquery, testTimeout, sessionProperties));
-                if (tearDownResult.getState() == State.SUCCESS) {
-                    break;
-                }
-                try {
-                    TimeUnit.MINUTES.sleep(1);
-                }
-                catch (InterruptedException e) {
-                    Thread.currentThread().interrupt();
-                    break;
-                }
-                retry++;
-            }
-            while (retry < testTeardownRetries);
-            // if teardown is not successful the query fails
-            queryResult = tearDownResult.getState() == State.SUCCESS ? queryResult : tearDownResult;
-        }
-        return queryResult;
+        return executeQueries(
+                queryPair.getTest(),
+                testPreQueryResults,
+                testGateway,
+                testUsername,
+                testPassword,
+                testTimeout,
+                testPostQueryResults,
+                testTeardownRetries);
     }
 
     private QueryResult executeQueryControl()
     {
-        Query query = queryPair.getControl();
+        return executeQueries(
+                queryPair.getControl(),
+                controlPreQueryResults,
+                controlGateway,
+                controlUsername,
+                controlPassword,
+                controlTimeout,
+                controlPostQueryResults,
+                controlTeardownRetries);
+    }
+
+    private QueryResult executeQueries(
```

Comment:
nit: `executeQuery`

Reply:
`executeQuery` doesn't seem right because we already have one. This is more like executing a query with setup and teardown, but I can't really come up with a name for it.